### DISCUSSION TITLE
Error messages when connecting to PLCSIM simulator

### DISCUSSION THREAD

**Discussion by denudge** [gopcua/opcua#583] [Q&A]
I try to run any of the examples against a PLCIM Advanced V 4.0 OPC Simulator, but I get the error message:

    opcua: uacp: message too large: 1347633490 > 65535 bytes
    
After tweaking my configuration to use the configs `opcua.ReceiveBufferSize(1347633490)` and `opcua.MaxMessageSize(1347633490)`, I get another error:

    unexpected EOF

I tried to debug the code and drilled it down to `TcpDial` -> `Handshake` -> `Send “HELF”` and then in the following `Receive()`, it gets 8 header bytes (string "HMMTRASP"), which is parsed to a header having "HMM" message type with the mentioned size of 1347633490. Then, when trying to read the message body with `ReadAtLeast()` using this size, only 37 bytes are read, followed by an EOF.

Maybe this is a misconfiguration issue of the simulator, but I actually don't know where to start.
I hope that someone could give me any hint.

Thank you in advance
Mathias

**Comment by magiconair** [comment]
Which version are you using? We recently added client side chunking which should support sending larger messages.

**Reply by denudge** [reply]
I am using version 0.3.4

**Comment by kung-foo** [comment]
Are you sure you are connecting to an OPC-UA endpoint? I find it unlikely that any server responds with a 1GB message.

**Reply by denudge** [reply]
Yes, that sounds really weird, I totally agree.
There is only one setting for the port in PLCSIM, it is called "runtime manager port". The description in the manual says "opens a port on the local machine", for whatever that means. But if I change this port to something else or even disable it (there is a checkbox aside), no connection is started at all. Then, the error differs between "connect: connection timed out" or "read: connection reset by peer" depending on whether I use the simulated ethernet address or just "localhost".
